File: src/main/java/com/ziwok/airticketsystem/api/model/Passenger.java
package com.ziwok.airticketsystem.api.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "PASSENGERS")
public class Passenger {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer passengerId;

    private String firstName;

    private String lastName;

    @Column(nullable = false)
    private LocalDate dateOfBirth;

    @Column(unique = true)
    private String passportNumber;

    private String nationality;
}

File: internal/entity/passenger.go
package entity

import (
	"time"
)

// Passenger represents a passenger record.
type Passenger struct {
	PassengerID int       `json:"passenger_id"`
	Name        string    `json:"name"`
	PhoneNumber string    `json:"phone_number"`
	Email       string    `json:"email"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
